docs/design/ttl_scans.md

```diff
@@ -0,0 +1,31 @@
+# TTL scans
+
+## Summary
+
+Add an option to automatically delete old starboard scans, in this first version focus on vulnerability scans but in the long run we could add similar functionality to other reports as well.
+
+## Motivation
+
+In [537](https://github.com/aquasecurity/starboard/issues/537) we talk about a need to run nightly vulnerability scans of CVE:s.
+This way we can make sure to get new CVE reports for long time running pods as well.
+
+## Proposal
+
+Add a feature feature flag of the operator, for example `--ttlVulnerabilityReport 24h`, this way we can add other ttl values for other reports as well.
+Adding this flag would add a spec/annotation to the generated VulnerabilityReport that we can look for.
+
+Using similar functionality as defined in [ttlAfterFinished](https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/) for cronjob seems to be the most versatile option.
+
+Just like the [ttlAfterFinished controller](https://github.com/kubernetes/kubernetes/blob/2a821d787b7f21fbb399b0a0bf6d3fa0c79f4293/pkg/controller/ttlafterfinished/ttlafterfinished_controller.go) we should add it as a separate controller.
```

Comment:
Just to confirm that we are talking about VulnerabilityReport controller here, right? And later on we could watch other kinds of security reports in a similar way.

Reply:
Yeah. I think the idea here is to have one ttl controller for all reports, not taking each existing controller and adding ttl to them.

Reply:
Just like @kfox1111 writes this is a seperate controller. For now I think we should focus on VulnerabilityReport but In the long run I think this might be useful for other reports as well. For example: https://github.com/aquasecurity/starboard/pull/830

This is also why my idea is to use `--ttlVulnerabilityReport 24h` and not `ttlReport 24h` since then we will be able to configure different values for different reports.